Refactor MTE-725 [v120] Run XCUITests on native m1

clarmso opened the pull request:
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/MTE-725)

## :bulb: Description
https://github.com/mozilla-mobile/firefox-ios/pull/16602 allows us to use native M1 binaries to run Firefox iOS instead of relying on Rosetta emulation. Now we can remove the M1-Rosetta workarounds. In addition, since the tests are running faster (saving approx 1/6 of time) there are some tests that fail due to the app not ready to be tapped. Some additional `mozWait*` have been added.

## :pencil: Checklist
You have to check all boxes before merging
- [X] Filled in the above information (tickets numbers and description of your work)
- [X] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed I updated documentation / comments for complex code and public methods

dragosb01 reviewed (approved):
@clarmso changes look good to me. Only one point i want to make. I have refactored ReaderModeTests and renamed the suite to ReadingListTests. I see you have some changes on this area and some conflicts may occur depending which PR is merged first, mine or yours. Also i see one update on HistoryTests that could cause conflicts. Here i think my version of updates should be kept cause i have addressed some comments from @isabelrios .

clarmso commented:
> @clarmso changes look good to me. Only one point i want to make. I have refactored ReaderModeTests and renamed the suite to ReadingListTests. I see you have some changes on this area and some conflicts may occur depending which PR is merged first, mine or yours. Also i see one update on HistoryTests that could cause conflicts. Here i think my version of updates should be kept cause i have addressed some comments from @isabelrios .

@dragosb01 Please merge your changes first. I can rebase and integrate your changes into mine.

clarmso commented:
Let me open a new PR. Some file names have had significant updates.
